src/common/io.ts

```diff
@@ -1000,6 +1004,72 @@ export class ARIOReadable implements AoARIORead, ArNSNameResolver {
       type: nameData.type,
     };
   }
+
+  /**
+   * Get all ARNS names associated with an address using the provided ANT registry address.
+   *
+   * By default it will use the mainnet ANT registry address.
+   *
+   * @param {Object} params - The parameters for fetching ARNS names
+   * @param {string} params.address - The address to fetch the ARNS names for
+   * @returns {Promise<AoArNSNameData[]>} The ARNS names associated with the address
+   */
+  async getArNSRecordsForAddress(
+    params: PaginationParams<AoArNSNameDataWithName> & {
+      antRegistryId?: string;
+      address: WalletAddress;
+    },
+  ): Promise<
+    PaginationResult<AoArNSNameDataWithName & { role: 'controller' | 'owner' }>
+  > {
+    const { antRegistryId = ANT_REGISTRY_ID, address } = params;
+    // create the ant registry
+    const antRegistry = ANTRegistry.init({
+      process: new AOProcess({
+        ao: this.process.ao,
+        processId: antRegistryId,
+      }),
+    });
```

Comment:
need to think about this more - if the list ever changes than we break our pagination logic, we likely need to return the processIds in the result, so it can be called recursively to respect pagination.